### ISSUE TITLE
emqttd_cm and emqttd_sm demonitor

### ISSUE THREAD

**Issue by congleetea** [emqtt/emqttd#1340]
In [emqttd_cm:erase_monitor/2](https://github.com/emqtt/emqttd/blob/master/src/emqttd_cm.erl#L154) and [emqttd_sm:erase_monitor/2](https://github.com/emqtt/emqttd/blob/master/src/emqttd_sm.erl#L312) 

```
erase_monitor(MRef, State = #state{monitors = Monitors}) ->
    State#state{monitors = dict:erase(MRef, Monitors)}.
```

I think it better if demonitor MRef before dict:erase, like this:

```
erase_monitor(MRef, State = #state{monitors = Monitors}) ->
    erlang:demonitor(MRef),
    State#state{monitors = dict:erase(MRef, Monitors)}.
```

**Comment by emqplus** [comment]
Hi @congleetea, the 'erase_monitor/2' function is usually called after the process received `{'DOWN', MonitorRef, _, _, _}` message. So is it necessary to `erlang:demonitor/1` the reference?

**Comment by congleetea** [comment]
I think it is more standard when  erlang:demonitor/1.

**Comment by emqplus** [comment]
@congleetea Fixed in 2.3.0 release. Close the issue now.
